app/controllers/import/github_controller.rb

```diff
@@ -0,0 +1,60 @@
+class Import::GithubController < Import::BaseController
+  before_filter :github_auth, except: :callback
+
+  rescue_from Octokit::Unauthorized, with: :github_unauthorized
+
+  def callback
+    token = client.get_token(params[:code])
+    current_user.github_access_token = token
+    current_user.save
+    redirect_to status_import_github_url
+  end
+
+  def status
+    @repos = client.repos
+    client.orgs.each do |org|
+      @repos += client.repos(org.login)
+    end
+
+    @already_added_projects = current_user.created_projects.where(import_type: "github")
+    already_added_projects_names = @already_added_projects.pluck(:import_source)
+
+    @repos.reject!{ |repo| already_added_projects_names.include? repo.full_name }
+  end
+
+  def jobs
+    jobs = current_user.created_projects.where(import_type: "github").to_json(only: [:id, :import_status])
```

Comment:
Line is too long. [106/80]<br>Prefer single-quoted strings when you don't need string interpolation or special symbols.